include/bpf/libbpf.h

```diff
@@ -943,6 +943,33 @@ bpf_program__flags(const struct bpf_program* prog);
 int
 bpf_program__set_flags(struct bpf_program* prog, __u32 flags);
 
+/**
+ * @brief Create BPF perfbuf manager.
+ *
+ * @param[in] map_fd File descriptor to perf event array map.
+ * @param[in] page_cnt Number of memory pages allocated for each per-CPU buffer. Currently unused.
+ * @param[in] sample_cb Pointer to perf buffer notification callback function.
+ * @param[in] lost_cb Function pointer for callback when record loss has occurred.
+ * @param[in] ctx User provided extra context passed into sample_cb and lost_cb.
+ * @return Pointer to perf buffer manager.
+ */
+LIBBPF_API struct perf_buffer*
+perf_buffer__new(
+    int map_fd,
+    size_t page_cnt,
+    perf_buffer_sample_fn sample_cb,
```

Comment:
Surprisingly these function pointer types and the ring buffer sample callback function pointer type are defined in api_internal.h. Not sure how applications will include this header without requiring api_internal.h. This could be a day - bug. Am I missing something here?